File: examples/scripts/queue_worker.lua
#!/usr/bin/env lua
-- examples/scripts/queue_worker.lua
-- Supervised worker loop. If it encounters a fatal error, it crashes ("let it fail"),
-- and the Clara supervisor restarts it based on its restart policy.

print("Worker started (PID: " .. (os.getenv("CLARA_RUN_ID") or "worker") .. ")")

-- Example polling loop or event consumer
-- In a real workflow, this might poll a Redis queue, subscribe to a websocket, etc.
for i = 1, 3 do
    print("Worker heartbeat tick: " .. i)
    -- Simulate sleep
    os.execute("sleep 1")
end

print("Worker cycle complete.")

File: examples/scripts/queue_worker.ts
#!/usr/bin/env bun
// examples/scripts/queue_worker.ts
// Supervised TypeScript worker loop executed with Bun.
// If it encounters a fatal unhandled error, it crashes ("let it fail"),
// and Clara supervisor automatically restarts it based on its restart policy.

import { runID, triggerID } from "../../scripts/clara";

console.log(`[Worker] Started queue processor (Run: ${runID()}, Trigger: ${triggerID()})`);

for (let i = 1; i <= 3; i++) {
  console.log(`[Worker] Heartbeat tick ${i}/3`);
  await new Promise((resolve) => setTimeout(resolve, 1000));
}

console.log("[Worker] Processing batch completed successfully.");
